deepchecks/nlp/context.py

```diff
@@ -326,14 +329,45 @@ def __init__(
             test_dataset = TextData.cast_to_dataset(test_dataset)
             if test_dataset.name is None:
                 test_dataset.name = 'Test'
+
         # If both dataset, validate they fit each other
         if train_dataset and test_dataset:
-            if test_dataset.has_label() and train_dataset.has_label() and not \
-                    train_dataset.validate_textdata_compatibility(test_dataset):
+            if (
+                test_dataset.has_label()
+                and train_dataset.has_label()
+                and not train_dataset.validate_textdata_compatibility(test_dataset)
+            ):
                 raise DatasetValidationError('train_dataset and test_dataset must share the same label and task type')
+
+            if (
+                train_dataset._properties is not None
+                and train_dataset._cat_properties is not None
+                and test_dataset._properties is not None
+                and test_dataset._cat_properties is not None
+            ):
```

Comment:
The if is redundant, the compare method should be able to menage situations where some of those parameters are none and the rest are not
e.x only test_dataset._cat_properties is none

Reply:
> the compare method should be able to menage situations where some of those parameters are none and the rest are not

hm/ I disagree, IMO, you are suggesting leaking `Context` specific logic into the `compare_dataframes` function, and that is not good

a task of the  `compare_dataframes` function is two compare two data frames, it should not care about anything more, making some of its parameters optional just will overcomplicate this function and makes it more confusing to understand

> e.x only test_dataset._cat_properties is none

sorry, do not understand/
`_properties` also can be none

its definition
```
_properties: t.Optional[pd.DataFrame] = None
```

Reply:
Main requirement - 
if one text data has 5 properties  with 1 of them being categorical and the other has 4 properties with 0 categorical (meaning _cat_properties is None) we still want the functionalities to work

Reply:
Specifically IMO the logic to handle this case should be inside the compare_dataframes function

Reply:
>if one text data has 5 properties with 1 of them being categorical and the other has 4 properties with 0 categorical (meaning _cat_properties is None) we still want the functionalities to work

oh, yeap, did not think about that, thanks!

>Specifically IMO the logic to handle this case should be inside the compare_dataframes function

moving that `if` statement into the `compare_dataframe` function just means that it will appear just a little bit later in the code

```
# NOW >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

if (
    train_dataset._properties is not None
    and test_dataset._properties is not None
):
    result = compare_dataframes(
        train=train_dataset._properties,
        test=test_dataset._properties,
        train_categorical_columns=train_dataset._cat_properties,
        test_categorical_columns=test_dataset._cat_properties
    )
    self._common_datasets_properties = result.common
    if result.difference is not None:
        get_logger().warning(
            'Train and Test datasets have different properties, '

# SUGGESTION >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

# returns none if one of the dataframes is none
result = compare_dataframes(
    train=train_dataset._properties,
    test=test_dataset._properties,
    train_categorical_columns=train_dataset._cat_properties,
    test_categorical_columns=test_dataset._cat_properties
)
if result is not None:
    self._common_datasets_properties = result.common
    if result.difference is not None:
        get_logger().warning(
            'Train and Test datasets have different properties, '
```